pkg/component/runtime/runtime_comm.go

```diff
@@ -83,7 +83,7 @@ func newRuntimeComm(logger *logger.Logger, listenAddr string, ca *authority.Cert
 		token:           token.String(),
 		cert:            pair,
 		checkinConn:     true,
-		checkinExpected: make(chan *proto.CheckinExpected),
+		checkinExpected: make(chan *proto.CheckinExpected, 1),
```

Comment:
Why the change here? I would prefer to not buffer expected configurations.

Reply:
so for this one the "command" runtime sends "CheckinExpected" while the beat for example failed to connect over gRPC and the V2 checkin call from the client is never called.
So this loop that reads from the unbuffered channel was never run
https://github.com/elastic/elastic-agent/blob/feature-arch-v2/pkg/component/runtime/runtime_comm.go#L184

and the CheckinExpected called from the "command" runtime is blocked forever until the components comes online and kicks off the "check-in" sequence.


We can either:
1. cache the latest "CheckinExpected" state until the component checks in over GRPs so we can send something back, or 
2. drop it on a floor, loosing some state that we suppose to synchronize on.
Think would have to redo the comms to keep the last state and not block on that chan

Reply:
Pushed another commit with the change in runtime_comms to addresses this particular issue according to the explanation above, implementing the caching of the latest state in the buffered channel of size 1.
https://github.com/elastic/elastic-agent/pull/1258/commits/d5f7de05a7a2ff9396ed73ddae6ea891bd945d88